app/models/cart.rb

```diff
@@ -24,16 +24,21 @@ def quantity(item_id)
 
   def increase_quantity(item_id)
     contents[item_id] += 1
-  end.to_s
+  end
 
   def decrease_quantity(item_id)
-    contents[item_id] -= 1
-  end.to_s
+    contents[item_id] -= 1 if contents[item_id] > 0 
+  end
 
   def find_items
     contents.keys.map {|item_id| Item.find(item_id)}
   end
 
+  def find_valid_items
+    items = subtotal.reject {|id, total| total == 0}
```

Comment:
Unused block argument - `id`. If it's necessary, use `_` or `_id` as an argument name to indicate that it won't be used.<br>Space between { and | missing.<br>Space missing inside }.